src/main/kotlin/io/sip3/salto/ce/rtpr/RtprHandler.kt

```diff
@@ -0,0 +1,175 @@
+            RtpReportPayload.SOURCE_RTCP -> PREFIX_RTCP
+            else -> throw IllegalArgumentException("Unsupported RTP report source: '${source}'")
+        }
+    }
+
+    open fun calculateMetrics(prefix: String, packet: Packet, report: RtpReportPayload) {
+        // Skip cumulative reports
+        if (report.cumulative) {
+            return;
+        }
+
+        val srcHost = packet.srcAddr.host
+        val dstHost = packet.dstAddr.host
+
+        report.apply {
+            val codecTag = codecName ?: payloadType.toString()
+            record(srcHost, dstHost, codecTag, prefix + EXPECTED_PACKETS, expectedPacketCount)
+            record(srcHost, dstHost, codecTag, prefix + LOST_PACKETS, lostPacketCount)
+            record(srcHost, dstHost, codecTag, prefix + REJECTED_PACKETS, rejectedPacketCount)
+
+            record(srcHost, dstHost, codecTag, prefix + JITTER, avgJitter)
+
+            record(srcHost, dstHost, codecTag, prefix + R_FACTOR, rFactor)
+            record(srcHost, dstHost, codecTag, prefix + MOS, mos)
+
+            Timer.builder(prefix + DURATION)
+                    .apply {
+                        srcHost?.let { tag("src_host", it) }
+                        dstHost?.let { tag("dst_host", it) }
+                        tag("codec", codecTag)
+                    }
+                    .register(Metrics.globalRegistry)
+                    .record(duration.toLong(), TimeUnit.MILLISECONDS)
+        }
+    }
+
+    open fun record(srcHost: String?, dstHost: String?, codecTag: String, name: String, value: Number) {
+        DistributionSummary.builder(name)
+                .apply {
+                    srcHost?.let { tag("src_host", it) }
+                    dstHost?.let { tag("dst_host", it) }
+                    tag("codec", codecTag)
+                }
+                .register(Metrics.globalRegistry)
+                .record(value.toDouble())
+    }
+
+    open fun writeToDatabase(prefix: String, packet: Packet, report: RtpReportPayload) {
+        val collection = if (report.cumulative) {
+            "${prefix}_index_" + timeSuffix.format(packet.timestamp)
+        } else {
+            "${prefix}_raw_" + timeSuffix.format(packet.timestamp)
+        }
+
+        val document = JsonObject().apply {
+            put("document", JsonObject().apply {
+                put("created_at", report.createdAt)
+                put("started_at", report.startedAt)
+
+                val src = packet.srcAddr
+                put("src_addr", src.addr)
+                put("src_port", src.port)
+                src.host?.let { put("src_host", it) }
+
+                val dst = packet.dstAddr
+                put("dst_addr", dst.addr)
+                put("dst_port", dst.port)
+                dst.host?.let { put("dst_host", it) }
+
+                put("payload_type", report.payloadType.toInt())
+                put("ssrc", report.ssrc)
+                report.callId?.let { put("call_id", it) }
```

Comment:
Wdyt if we will filter packets without call_id? For database at the moment packets without call_id is like rubish

Reply:
I mean to filter it within `handle` method

Reply:
I meant, let's do not save them to database. Because they just consume the disk space and never will be read.

Reply:
Let's skip it for now